src/applications/mhv/secure-messaging/tests/components/MessageListComponents/folderHeader.unit.spec.jsx

```diff
@@ -30,41 +38,275 @@ describe('FolderHeader component in custom folder', () => {
+      );
+      expect(filterInputElement.getAttribute('aria-label')).to.equal(
+        `Filter messages in ${
+          customFolder.name
+        } Enter information from one of these fields: to, from, message ID, or subject`,
+      );
+    });
+
+    it('does not render FilterBox w/o `threadCount` on CUSTOM FOLDER', () => {
+      const screen = setup(initialState, initialPath, null);
+      expect(screen.queryByTestId('search-form')).to.not.exist;
+    });
   });
 
-  it('must display valid custom folder name', async () => {
-    expect(screen.getByText('TEST2')).to.exist;
-    expect(screen.getByText(Folders.CUSTOM_FOLDER.desc)).to.exist;
+  describe('Folder Header component displays INBOX folder and children components', () => {
+    const initialInboxState = {
+      sm: {
+        folders: {
+          folder: inbox,
+          folderList,
+        },
+      },
+      user: {
+        profile: {
+          session: {
+            ssoe: true,
+          },
+        },
+      },
+    };
+
+    it('must display valid FOLDER name: INBOX', async () => {
+      const screen = setup(initialState, Paths.INBOX, threadList, inbox);
+      expect(screen.getByText(inbox.name, { selector: 'h1' })).to.exist;
+      expect(screen.queryByText(Folders.CUSTOM_FOLDER.desc)).to.not.exist;
+    });
+
+    it('renders FilterBox with `threadCount` in INBOX FOLDER', () => {
+      const screen = setup(initialInboxState, Paths.INBOX, threadList, inbox);
+      const filterBox = screen.getByTestId('search-form');
+      expect(filterBox).to.exist;
+
+      const filterHeader = filterBox.querySelector('h2');
+      expect(filterHeader.textContent).to.equal(
+        `Filter messages in ${inbox.name} `,
+      );
+
+      const filterInputElement = screen.queryByTestId('keyword-search-input');
+      expect(filterInputElement.getAttribute('label')).to.equal(
+        'Enter information from one of these fields: to, from, message ID, or subject',
+      );
+      expect(filterInputElement.getAttribute('aria-label')).to.equal(
+        `Filter messages in ${
+          inbox.name
+        } Enter information from one of these fields: to, from, message ID, or subject`,
+      );
+    });
+
+    it('does not render FilterBox w/o `threadCount` on INBOX FOLDER', () => {
+      const screen = setup(initialInboxState, Paths.INBOX, null, inbox);
+      expect(screen.queryByTestId('search-form')).to.not.exist;
+    });
   });
 
-  it('verifies page title tag for custom folder page', async () => {
-    await waitFor(() => {
-      expect(global.document.title).to.equal(
-        `TEST2 ${PageTitles.PAGE_TITLE_TAG}`,
+  describe('Folder Header component displays DRAFTS folder and children components', () => {
+    const initialDraftsState = {
+      sm: {
+        folders: {
+          folder: drafts,
+          folderList,
+        },
+      },
+      user: {
+        profile: {
+          session: {
+            ssoe: true,

```

Comment:
same question as above if this is needed

Reply:
Same explanation here as above.